app/sidebar.py

```diff
@@ -119,12 +119,16 @@ def make_sidebar(api: PtxboaAPI):
     st.session_state["scenario"] = f"{data_year} ({cost_scenario})"
 
     st.sidebar.subheader("Additional settings:")
-    st.session_state["secproc_co2"] = st.sidebar.radio(
-        "Carbon source:",
-        api.get_dimension("secproc_co2").index,
-        horizontal=True,
-        help=read_markdown_file("md/helptext_sidebar_carbon_source.md"),
-    )
+
+    # check if carbon is needed as input:
+    needs_co2 = check_if_input_is_needed(api, flow_code="CO2-G")
+    if needs_co2:
+        st.session_state["secproc_co2"] = st.sidebar.radio(
+            "Carbon source:",
+            api.get_dimension("secproc_co2").index,
+            horizontal=True,
+            help=read_markdown_file("md/helptext_sidebar_carbon_source.md"),
+        )

```

Comment:
If we would once switch the default chain, we could run into a problem because then st.session_state["secproc_co2"] is not defined then. Should we just set it to None if `needs_co2==False`?

Reply:
Good idea, I fixed this.